src/nupic/proto/LinkProto.capnp

```diff
@@ -10,4 +12,8 @@ struct LinkProto {
 
   destRegion @4 :Text;
   destInput @5 :Text;
+
+  outputElementType @6 :Int32; # NTA_BasicType: type of output elements in output array
+  outputElementCount @7 :UInt32; # number of output elements in output array
```

Comment:
Is `outputElementCount` necessary? I believe that when you deserialize `delayedOutputs` you can check how many elements there are.